### PR TITLE
Restructure package

### PR DESCRIPTION
**Pull request by steven-murray**
Fully restructured package.

Package structure created with `pyscaffold`. Adds following features:

* Versioning support
* Full package infrastructure (license, authors, etc.)
* Proper tests subdirectory
* Tox/travis support (not fully implemented in this PR)
* black formatting
* py3 compat
* flake8

Also, to ease installation, I've added `ssht` as a git submodule. I was able to get it compiling and installing without any need for a YAML file pointing to include directories. It also now compiles for free when doing `pip install .`. 

I have not yet been able to get `pyssht` to install to do the tests, so would be grateful for help on that front, @zacharymartinot .

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "ssht"]
```
Comment (plaplant) [inline .gitmodules:1]: Following the discussion on slack, we might just be able to change the url here.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# THIS SCRIPT IS SUPPOSED TO BE AN EXAMPLE. MODIFY IT ACCORDING TO YOUR NEEDS!
+
+sudo: false
+language: python
+virtualenv:
+  system_site_packages: false
+matrix:
+  fast_finish: true
+  include:
+    - python: 3.6
```
Comment (plaplant) [inline .travis.yml:12]: As with spin1_beam_model, I think it'd be good to build for py3.7 too, but that can be done in a separate PR when we've got full-fledged tests going.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
+==========
```
Comment (plaplant) [inline README.rst:1]: Is there a benefit to using RST vs. markdown? Just wondering why the change in readme format.

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+----------
+This package has `pre-commit` hooks set up. If developing this package, please
+`pip install pre-commit` and `pre-commit install`.
+
+Versioning
+----------
+This package uses `pyscaffold`, which in turn uses `setuptools_scm`, and manages
+versioning via Git tags. Versioning should use https://semver.org semantic versioning
+rules.
+>>>>>>> Stashed changes
```
Comment (plaplant) [inline README.rst]: Looks like a git diff marker sneaked through.

File: build_ssht_cffi.py [hunk 5]
```diff
@@ -1,147 +1,84 @@
-import os
```
Comment (plaplant) [inline build_ssht_cffi.py:1]: I can't seem to make a comment on the `__init__.py` file, so I'll ask the question here: Why do we need an `__init__.py` in the top level of the repo? Do we not use the traditional package organization here?

File: build_ssht_cffi.py [hunk 6]
```diff
@@ -1,147 +1,84 @@
...
+import os
 import shutil
+import subprocess
+import tempfile
 
-import yaml
 from cffi import FFI
 
 ffi = FFI()
+this_dir = ""  # os.path.dirname(os.path.abspath(__file__))
```
Comment (plaplant) [inline build_ssht_cffi.py]: We should remove the comment if we're not getting the path this way anymore.

File: build_ssht_cffi.py [hunk 7]
```diff
@@ -1,147 +1,84 @@
...
+    inc_string,
+    sources=ssht_src_files,
+    libraries=["fftw3"],
+    include_dirs=include_dirs,
+    extra_link_args=extra_link_args,
+    extra_compile_args=makeopts,
+)
+
+
+# def build():
```
Comment (plaplant) [inline build_ssht_cffi.py]: We should remove this function rather than have it commented out.

File: setup.cfg [hunk 8]
```diff
@@ -0,0 +1,119 @@
+# This file is used to configure your project.
+# Read more about the various options under:
+# http://setuptools.readthedocs.io/en/latest/setuptools.html#configuring-setup-using-setup-cfg-files
+
+[metadata]
+name = ssht_numba
+description = Add a short description here!
```
Comment (plaplant) [inline setup.cfg]: We should add a description for the package.

File: tests/test_numba_wrappers.py [hunk 9]
```diff
@@ -0,0 +1,125 @@
+"""
+Tests the numba wrapper functions against the equivalent pyssht functions
+using random input data.
```
Comment (plaplant) [inline tests/test_numba_wrappers.py:3]: I'm okay with using random input as long as there are no unintended side-effects related to the stochasticity. It might be good to have some tests that use known input for some things (if that's not done already).

**Review by plaplant — COMMENTED** [review]
Thanks @steven-murray! I have a few small questions, and ones similar to the spin1_beam_model PR that I didn't explicitly post again.

File: .gitmodules (hunk 1 above)
Comment (steven-murray) [reply .gitmodules:1]: Yeah, I'll try that.

File: README.rst (hunk 3 above)
Comment (steven-murray) [reply README.rst:1]: I've always used RST, because then you can directly include the README in your Sphinx docs without writing weird conversion scripts, or having redundant text. We can use md if you like.

File: README.rst (hunk 4 above)
Comment (steven-murray) [reply README.rst]: Woops... will fix.

File: build_ssht_cffi.py (hunk 5 above)
Comment (steven-murray) [reply build_ssht_cffi.py:1]: It's an artifact of my convoluted process of getting the restructure done (involving copying over the new package into the old repo...). I'll delete it.

File: build_ssht_cffi.py (hunk 6 above)
Comment (steven-murray) [reply build_ssht_cffi.py]: Agreed. removed.

File: build_ssht_cffi.py (hunk 7 above)
Comment (steven-murray) [reply build_ssht_cffi.py]: Also agreed and removed.

File: setup.cfg (hunk 8 above)
Comment (steven-murray) [reply setup.cfg]: Done. And removed my name as Author... that was automatic from pyscaffold

File: tests/test_numba_wrappers.py (hunk 9 above)
Comment (steven-murray) [reply tests/test_numba_wrappers.py:3]: I think this is a good idea, and we should make an issue for it (I didn't touch the tests in this PR other than to do formatting etc.)

File: README.rst (hunk 3 above)
Comment (plaplant) [reply README.rst:1]: Nope, I'm fine with RST, was just wondering about the motivation for changing the readme format.

**Review by plaplant — APPROVED** [review]
Thanks for this @steven-murray! I'm in favor of merging this in now, and we can address lingering pyssht issues in another PR.
